apps/studio/src/hooks/useNavigationEffect.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from "react"
+import { useRouter } from "next/router"
+
+interface UseNavigationEffectProps {
+  isOpen?: boolean
+  isDirty?: boolean
+  callback: (url: string) => void
+}
+export const useNavigationEffect = ({
```

Comment:
nit: kekw I know what this does but I think not very obvious, maybe can add a comment describing this?